crates/lang/src/parse.rs

```diff
@@ -549,13 +549,178 @@ fn seek_to<'a, R: pest::RuleType>(
     None
 }
 
+#[derive(Debug, PartialEq)]
+pub struct RemoteDbInfo {
+    addrs: Vec<RemoteAddr>,
+    username: Option<String>,
+    password: Option<String>,
+    database_name: String,
+    table_name: String,
+}
+
+#[derive(Debug, Default, PartialEq)]
+pub struct RemoteAddr {
+    ip_addr: Option<IpAddr>,
+    host_name: Option<String>,
+    port: Option<u16>,
+}
+
+impl Default for RemoteDbInfo {
+    fn default() -> Self {
+        Self {
+            addrs: vec![],
+            username: None,
+            password: None,
+            database_name: "default".to_string(),
+            table_name: "".to_string(),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub enum QueryFormat {
```

Comment:
the naming of QueryFormat is too big. something like TablePlaceKind or TableHostKind may be better.